var/spack/repos/builtin/packages/zlib/package.py

```diff
@@ -25,27 +25,42 @@ class Zlib(Package):
             description='Produce position-independent code (for shared libs)')
     variant('shared', default=True,
             description='Enables the build of shared libraries.')
+    variant('debug',   default=False,
+            description='Build debug code (-g -O0)')
```

Comment:
Is this to build debug code for `zlib` only? I am just trying to understand why not using `cflags=` syntax in the spec.

Reply:
Yes, it affects zlib only. It also removes flags that conflict with it (so `zlib cflags=-O3 +debug` still does a debug `-g -O0` build.

Reply:
(comment: no change required) This might be worth discussing but probably shouldn't hold the PR up: this could result in multiple equivalent zlib installs with different `cflags` settings (if all of them are built with `+debug`).